### PR TITLE
Convert history calls to use new websocket endpoint

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Switch history calls to use new websocket endpoint which reduces the contention for the http connection and improves the response time.

~~Requires https://github.com/home-assistant/core/pull/71688~~
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bdraco** [comment]
Unrelated to this PR since its doing it with the rest api, I noticed that history requests the full past 24 hours every 10 seconds when the card is open and firing events.  Logbook only requests what has changed since the last window.  I think the cache might be broken

**Comment by bdraco** [comment]
This also solves the delay I was having loading the energy panel with a lot of sensors

File: src/data/history.ts [hunk 1]
```diff
@@ -212,65 +268,68 @@ const equalState = (obj1: LineChartState, obj2: LineChartState) =>
...
+            processedState.attributes![attr] = state.a[attr];
           }
         }
       } else {
-        processedState = state;
+        processedState = {
+          state: state.s,
+          last_changed: state.lc * 1000,
+          attributes: {},
+        };
```
Comment (bdraco) [inline src/data/history.ts:332]: I could implement the fallback we discussed here or wait for a followup PR?

https://github.com/home-assistant/core/pull/71843#issuecomment-1126852642

Comment (bdraco) [reply src/data/history.ts:332]: https://github.com/home-assistant/core/pull/71886 Created this one.

Comment (bdraco) [reply src/data/history.ts]: Actually it was just as easy to make the code backwards compatible so I did that.

**Review by balloob — APPROVED** [review]

**Pull request merged**